pyEpiabm/pyEpiabm/core/cell.py

```diff
@@ -70,16 +71,24 @@ def add_microcells(self, n):
         for _ in range(n):
             self.microcells.append(Microcell(self))
 
-    def set_id(self, id: float):
+    def set_id(self, id: str):
```

Comment:
Open question - should we check for uniqueness in this method? It would not be a big time cost, so I think it might make sense to check there is not a duplicate ID in the existing list of cell IDs. This would likely throw an error (or at least be logged) if it is a duplicate. You will also need to test this behavior I'm afraid (and replicate for other classes)

Reply:
Yes, I will do this

Reply:
Checked duplicates for microcell, household and person, but not sure how to do this for cell without adding an extra parameter to the set_id function containing a list of all other cells, which does not seem like a good idea

Reply:
Actually, we are going to take the above approach (passing a list of cells to the set_id function) as this is done similarly in another function, but let us know if you would rather an alternative approach

Reply:
I see. Yes I don't love this - where else do we do something similar? I am almost tempted to store the population as an attribute of the cell (like the cell is an attribute of the microcell) to avoid this. But if you have seen it somewhere else then I suppose it is not a big deal, so happy to work with this

Reply:
In cell.py, the function find_nearby_cells uses a list of cells